proposals/calls-to-action-for-better-patch-adoption.md

```diff
@@ -0,0 +1,126 @@
+
+### Title
+
+<!-- A short, pithy title for the proposal. -->
+Calls to Action for Better Patch Adoption
+
+### Short description
+
+If patches are available for upstream vulnerabilities, why don't more projects upgrade? What are the simplest systemic improvements that could make a difference at scale?
+
+### Author(s)
+
+<!-- Put your GitHub username(s) here. The proposal author(s) will "own" the proposal and will be able to accept future PRs that change it. -->
+[GitHub: lucasgonze](https://github.com/lucasgonze/)
+
+### Proposal body
+
+<!-- Explain your proposal. Add as much as you want, within reason! -->
+Critical vulnerabilities often linger in deployed software long past the point when a patch is available. Maven estimates that [35% of Log4J downloads continue to pull the version with the world-famous vulnerability](https://www.sonatype.com/resources/log4j-vulnerability-resource-center?smtNoRedir=1). Why?
+
+More importantly, what are the most promising remediations? 
+
+Let's fix that. Let's go find out. I propose research into the reasons why patches aren't being applied.
+
+#### Go-To-Market and Roadmap
+
+0. Gather a list of potential remediations
+ - Survey prior research
+ - Brainstorm
+ - Document steps to apply each remediation
+
+1. Gain access to repositories
+
+  - Identify impactful repos
+    - Avoid toy projects, unreleased code, projects with low adoption
+    - Find some commercial code bases that use open source libraries
+  - Gain access to repos
+    - For proprietary code bases, may need to cultivate relationships in order to have access. OSPOs may be able to help.
+    - Focus on open source projects, because we have easy access to the code
+  - Span source-code hosting tools
+    - GitHub
+    - GitLab
+    - Independent
+
+2. Identify open vulnerabilities
+
+  - Gather a manifest
+  - Compare manifest to CVE catalog
+
+3. Categorize causes of vulnerabilities
+
+  - Not using Dependabot at all
+  - Not applying auto-upgrades
+  - Auto-upgrades blocked on breaking changes in upgrade path
+  - Inadequate infrastructure (tooling, test coverage, QA) to perform auto-upgrades
+  - Upgrades applied in dev but not released to production
+  - Distrust in reliability of supposed non-breaking changes
+  - Pinned version in package lock file
+
+4. Document findings
+  - Map remediations to causes and identify most promising options
+  - Report to OpenSSF
```

Comment:
Have you chatted with folks at OpenSSF to see how this work might fit in with their priorities / existing work?

Reply:
Absolutely. This idea came up in the context of an OpenSSF conversation.